wallet/src/de/schildbach/wallet/ui/InviteHandlerViewModel.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 Dash Core Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.schildbach.wallet.ui
+
+import android.app.Application
+import android.content.Intent
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.google.firebase.dynamiclinks.FirebaseDynamicLinks
+import de.schildbach.wallet.data.InvitationLinkData
+import de.schildbach.wallet.livedata.Resource
+import de.schildbach.wallet.ui.dashpay.BaseProfileViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.slf4j.LoggerFactory
+
+class InviteHandlerViewModel(application: Application) : BaseProfileViewModel(application) {
+
+    private val log = LoggerFactory.getLogger(InviteHandlerViewModel::class.java)
+
+    val inviteData = MutableLiveData<Resource<InvitationLinkData>>()
+
+    val invite: InvitationLinkData
+        get() = inviteData.value!!.data!!
+
+    fun handleInvite(intent: Intent) {
+        FirebaseDynamicLinks.getInstance().getDynamicLink(intent).addOnSuccessListener {
+            val link = it?.link
+            if (link != null && InvitationLinkData.isValid(link)) {
+                log.debug("received invite $link")
+                val invite = InvitationLinkData(link, false)
+                handleInvite(invite)
+            } else {
+                log.debug("invalid invite ignored")
+            }
+        }
+    }
+
+    fun handleInvite(invite: InvitationLinkData) {
+        inviteData.value = Resource.loading()
+        viewModelScope.launch(Dispatchers.IO) {
+            try {
+                // don't validate onboarding invites (they will be validated afterwards)
+                invite.validation = (walletApplication.wallet == null) || platformRepo.validateInvitation(invite)
```

Comment:
This approach is interesting. Mainly I am making a note for later reference here.

Reply:
yeah, I'm not sure if it is a good approach...